ethereum_transaction_data.go

```diff
@@ -20,121 +20,61 @@ package hedera
  *
  */
 
-import (
-	"encoding/json"
-
-	"github.com/pkg/errors"
-
-	"github.com/ethereum/go-ethereum/core/types"
-	"github.com/ethereum/go-ethereum/rlp"
-)
-
 // Represents the data of an Ethereum transaction.
 type EthereumTransactionData struct {
-	eip1559 *types.DynamicFeeTx
-	legacy  *types.LegacyTx
+	eip1559 *EthereumEIP1559Transaction
+	legacy  *EthereumLegacyTransaction
 }
 
 // EthereumTransactionDataFromBytes constructs an EthereumTransactionData from a raw byte array.
 func EthereumTransactionDataFromBytes(b []byte) (*EthereumTransactionData, error) {
 	var transactionData EthereumTransactionData
-	if b[0] == 2 {
-		byt := b
-		byt = append(byt[:0], byt[0+1:]...)
-		err := rlp.DecodeBytes(byt, &transactionData.eip1559)
+	if b[0] == 0x02 {
+		eip1559, err := EthereumEIP1559TransactionFromBytes(b)
 		if err != nil {
 			return nil, err
 		}
 
+		transactionData.eip1559 = eip1559
 		return &transactionData, nil
 	}
 
-	err := rlp.DecodeBytes(b, &transactionData.legacy)
+	legacy, err := EthereumLegacyTransactionFromBytes(b)
 	if err != nil {
 		return nil, err
 	}
 
+	transactionData.legacy = legacy
 	return &transactionData, nil
 }
 
 // ToBytes returns the raw bytes of the Ethereum transaction.
-func (ethereumTxData *EthereumTransactionData) ToBytes() ([]byte, error) {
-	var byt []byte
-	var err error
-	if ethereumTxData.eip1559 != nil {
-		byt, err = rlp.EncodeToBytes(ethereumTxData.eip1559)
-		if err != nil {
-			return []byte{}, err
-		}
-		byt = append([]byte{2}, byt...)
-
-		return byt, nil
+func (txData *EthereumTransactionData) ToBytes() ([]byte, error) {
+	if txData.eip1559 != nil {
+		return txData.eip1559.ToBytes()
 	}
 
-	byt, err = rlp.EncodeToBytes(ethereumTxData.legacy)
-	if err != nil {
-		return []byte{}, err
+	if txData.legacy != nil {
+		return txData.legacy.ToBytes()
 	}
 
-	return byt, nil
+	return nil, nil
 }
 
 func (ethereumTxData *EthereumTransactionData) _GetData() []byte {
 	if ethereumTxData.eip1559 != nil {
-		return ethereumTxData.eip1559.Data
+		return ethereumTxData.eip1559.CallData
 	}
 
-	return ethereumTxData.legacy.Data
+	return ethereumTxData.legacy.CallData
 }
 
 func (ethereumTxData *EthereumTransactionData) _SetData(data []byte) *EthereumTransactionData {
 	if ethereumTxData.eip1559 != nil {
-		ethereumTxData.eip1559.Data = data
+		ethereumTxData.eip1559.CallData = data
 		return ethereumTxData
 	}
 
-	ethereumTxData.legacy.Data = data
+	ethereumTxData.legacy.CallData = data
 	return ethereumTxData
 }
-
-// ToJson returns a JSON representation of the Ethereum transaction.
-func (ethereumTxData *EthereumTransactionData) ToJson() ([]byte, error) {
```

Comment:
We should check if these json related funcs were not requested from user and if it's safe to remove them.

Reply:
Do the new struct types break the to/fromJson funcs? The old implementation looks like it would work.